### ISSUE TITLE
universal ctags and c++ atributes

### ISSUE THREAD

**Issue by andrejlevkovitch** [universal-ctags/ctags#2364]
The name of the parser: `universal ctags`

The command line you used to run ctags:

```
$ ctags main.cpp
```

The content of input file:

```c++
// main.cpp
#include <cstdlib>
#include <iostream>

void foo();

int main([[maybe_unused]]int argc, [[maybe_unused]]char *argv[]) {
  int alpha;
  int bravo;
  int charlie;

  return EXIT_SUCCESS;
}
```

The tags output you are not satisfied with:

```
!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_MODE	u-ctags	/u-ctags or e-ctags/
!_TAG_PROGRAM_AUTHOR	Universal Ctags Team	//
!_TAG_PROGRAM_NAME	Universal Ctags	/Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL	https://ctags.io/	/official site/
!_TAG_PROGRAM_VERSION	0.0.0	//
alpha	main.cpp	/^  int alpha;$/;"	v	typeref:typename:int
bravo	main.cpp	/^  int bravo;$/;"	v	typeref:typename:int
charlie	main.cpp	/^  int charlie;$/;"	v	typeref:typename:int
foo	main.cpp	/^void foo() {$/;"	f	typeref:typename:void
```

The tags output you expect:

```
!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_MODE	u-ctags	/u-ctags or e-ctags/
!_TAG_PROGRAM_AUTHOR	Universal Ctags Team	//
!_TAG_PROGRAM_NAME	Universal Ctags	/Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL	https://ctags.io/	/official site/
!_TAG_PROGRAM_VERSION	0.0.0	//
foo	main.cpp	/^void foo() {$/;"	f	typeref:typename:void
main	main.cpp	/^int main(int argc, char *argv[]) {$/;"	f	typeref:typename:int
```

The version of ctags:

```
$ ctags --version
Universal Ctags 0.0.0, Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Jul 13 2019, 19:03:10
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +packcc
```

How do you get ctags binary: build it localy by `gcc`

**Comment by masatake** [comment]
Thank you for reporting.

I think we can skip [[foo]] as we do for __attribute__(((foo))__.

**Comment by masatake** [comment]
@pragmaware, could you look into this?

**Comment by masatake** [comment]
If possible, plesae, try #2371.

It seems that the maintainer of C++ parser may be in busy.
So I made a temporary fix, and make the pull request .

Merging to the master branch may be done by @pragmaware, the maintainer of the C++ parser.

**Comment by pragmaware** [comment]
Ow... you preceeded me by 10 minutes :D

**Comment by masatake** [comment]
Wow. Of course, your one is much better. I withdraw my pull request. I will study your code.

**Comment by pragmaware** [comment]
No wait... let me think a moment. Your code is far shorter and simpler than mine.

**Comment by pragmaware** [comment]
@masatake : I kind of prefer your idea because it's far shorter and simpler. I think I'll also withdraw and make a pull request based on yours.

**Comment by masatake** [comment]
With your approach, we can store the values of attributes to a field as you did for "static", "inline", ...
I like it.

**Comment by pragmaware** [comment]
@masatake: your approach can be extended to handle that too. See #2373 .

**Comment by masatake** [comment]
Oh, I see. The new pull request allows us  to inspect inside `[[`, `]]` when we want to do so:-P.

**Comment by pragmaware** [comment]
Should be fixed by #2373
